src/main/java/tr/edu/ogu/ceng/service/InternshipRegistryService.java

```diff
@@ -35,11 +37,11 @@ public InternshipRegistryDto updateInternshipRegistry(InternshipRegistryDto inte
 		if (!internshipRegistryRepository.existsById(internshipRegistry.getId()))
 			throw new EntityNotFoundException("Internship Registry not found!");
 
-		Timestamp localDateTime = new Timestamp(System.currentTimeMillis());
-		internshipRegistry.setCreateDate(internshipRegistry.getCreateDate());
-		internshipRegistry.setUpdateDate(localDateTime);
+		internshipRegistry.setCreateDate(internshipRegistryRepository.getById(internshipRegistry.getId()).getCreateDate());
```

Comment:
Burada createDate'i tekrar set etmek yerine dto'ya ekleyebilirsin diye düşünüyorum

Reply:
Update işlemi sonucu createDate değişkeninin değişmemesi gerekmez mi? Bu sebeple, dto'da taşınmasının bir anlamı olmadığını düşünüyorum. Veri tabanında tutulan ilk oluşturulma değeri korunmalıdır.

Reply:
Response'da da dto dönüyoruz ve frontend'te createDate updateDate gibi verilere ihtiyaç olabilir. Buradaki asıl sorun requestDto ve responseDto'nun ayrı ayrı oluşturulması gerekiyor fakat bu şekilde yapmadık. Kullanıcıdan request ile createDate ve updateDate almamalıyız doğru söylüyorsun fakat döndürdüğümüz response'da bunlara ihtiyacımız olabilir diye düşünüyorum. Şimdilik Dto lar request ve response olarak ayrılmadığı için ve diğer dto'ları incelediğimde de createDate-updateDate içerdiklerini gördüğüm için bunların şuanki mevcut Dto'larda var olması gerektiğini düşünüyorum

Reply:
Response ve Request için DTO'ların ayrılmasından bağımsız olarak servis katmanında istemciden gelen objenin doğrudan veritabanına kaydedilmesi yerine createDate değerinin istemciden ne gelirse gelsin dikkate alınmaması, updateDate değerinin ise istemciden ne gelirse gelsin sistem tarih ve saati ile güncellenmesi ile veritabanına kaydedilmesi gerektiğini söyleyebiliriz.

Reply:
@Melisaozzel istemcinin aldığı createDate değerini doğru olarak geri göndereceğine güvenemeyiz.